spokestack/asr/cloud_client.py

```diff
@@ -66,18 +66,30 @@ def __init__(
         self._idle_timeout = idle_timeout
         self._idle_count: int = 0
 
-    def __call__(self, audio: np.ndarray, n_best: int = 1) -> List[str]:
+    def __call__(self, audio: Union[bytes, np.ndarray], limit: int = 1) -> List[str]:
         """ Audio to text interface for the cloud client
 
         Args:
-            audio (np.ndarray): np.int16 array of audio
-            n_best (int): number of predictions to return
+            audio (bytes|np.ndarray): input audio can be in the form of
+                                      bytes or np.float, np.int16 array with
+                                      conversions handled. other types with produce
+                                      a TypeError
+            limit (int): number of predictions to return
 
-        Returns: n_best list of transcripts
+        Returns: list of transcripts, and their confidence values of size limit
 
         """
+        if isinstance(audio, bytes):
+            audio = np.frombuffer(audio, np.int16)
+        elif np.issubdtype(audio.dtype, np.int16):
+            audio = audio
```

Comment:
no need to assign audio to itself here, just put this as the last clause and invert it